Apsitvarkom.Api/Controllers/CleaningEventController.cs

```diff
@@ -84,16 +87,30 @@ public async Task<ActionResult<CleaningEventResponse>> GetById([FromQuery] Objec
 
     [ProducesResponseType(StatusCodes.Status201Created)]
     [ProducesResponseType(StatusCodes.Status400BadRequest, Type = typeof(List<string>))]
+    [ProducesResponseType(StatusCodes.Status404NotFound)]
     [ProducesResponseType(StatusCodes.Status500InternalServerError)]
     [HttpPost("Create")]
     public async Task<ActionResult<CleaningEventResponse>> Create(CleaningEventCreateRequest cleaningEventCreateRequest)
     {
         var validationResult = await _cleaningEventCreateValidator.ValidateAsync(cleaningEventCreateRequest);
         if (!validationResult.IsValid) return BadRequest(validationResult.Errors.Select(e => e.ErrorMessage).ToList());
 
+        PollutedLocation? location;
+        try
+        {
+            location = await _pollutedLocationRepository.GetByPropertyAsync(x => x.Id == cleaningEventCreateRequest.PollutedLocationId);
```

Comment:
we don't need that object, we just need to make sure the PollutedLocation exists. we could instead use a 'lighter' (it only has to return a boolean instead of whole instance) method - `ExistsByPropertyAsync`, which only checks if such instance can be found in the database, but doesn't return any of its properties (because we don't really need them - we have the `PollutedLocationId` from the `CleaningEventCreateRequest`

Reply:
check out Delete action to see how it can be used - it's similar, only minor adjustments will be necessary.